--- /usr/bin/cpanflute2.orig	2009-04-02 21:17:47.000000000 +0900
+++ /usr/bin/cpanflute2	2009-04-02 21:25:15.000000000 +0900
@@ -288,17 +288,20 @@
 
 my $makefile_pl = qq{CFLAGS="\$RPM_OPT_FLAGS" %{__perl} Makefile.PL < /dev/null};
 my $make_install = qq{make pure_install PERL_INSTALL_ROOT=\$RPM_BUILD_ROOT};
-
+my $make;
 if ($use_module_build) {
-  $makefile_pl = qq{CFLAGS="\$RPM_OPT_FLAGS" %{__perl} Makefile.PL destdir=\$RPM_BUILD_ROOT $installdirs < /dev/null};
+  $makefile_pl = qq{CFLAGS="\$RPM_OPT_FLAGS" %{__perl} Build.PL destdir=\$RPM_BUILD_ROOT $installdirs < /dev/null};
+  $make_install = qq{./Build pure_install PERL_INSTALL_ROOT=\$RPM_BUILD_ROOT};
+  $make = "./Build %{?_smp_mflags} OPTIMIZE=\"\$RPM_OPT_FLAGS\"";
 }
 else {
   $makefile_pl = qq{CFLAGS="\$RPM_OPT_FLAGS" %{__perl} Makefile.PL $installdirs};
+  $make = "make %{?_smp_mflags} OPTIMIZE=\"\$RPM_OPT_FLAGS\"";
 }
 
 $spec->build(<<EOB);
 $makefile_pl
-make %{?_smp_mflags} OPTIMIZE="\$RPM_OPT_FLAGS"
+$make
 EOB
 
 $spec->clean('rm -rf $RPM_BUILD_ROOT');
